tests/Kernel/ContactRenderTest.php

```diff
@@ -0,0 +1,245 @@
+      'oe_theme_content_entity_contact',
+    ]);
+
+    $this->contactStorage = $this->container->get('entity_type.manager')->getStorage('oe_contact');
+    $this->contactViewBuilder = $this->container->get('entity_type.manager')->getViewBuilder('oe_contact');
+  }
+
+  /**
+   * Test a contact being rendered using full view.
+   */
+  public function testFullView(): void {
+    // Create Contact entity with required values only.
+    $contact = $this->createContactEntity();
+    $name = $contact->getName();
+
+    $build = $this->contactViewBuilder->view($contact, 'full');
+    $html = $this->renderRoot($build);
+    $crawler = new Crawler($html);
+
+    // Assert that empty values don't exist.
+    $this->assertEmpty($crawler->filter('.ecl-editor'));
+    $this->assertEmpty($crawler->filter('.ecl-description-list'));
+    $this->assertEmpty($crawler->filter('figure.ecl-media-container'));
+    $this->assertEmpty($crawler->filter('.ecl-u-border-top.ecl-u-border-bottom.ecl-u-border-color-grey-15.ecl-u-mt-s.ecl-u-pt-l.ecl-u-pb-l'));
+
+    // Assert Contact title.
+    $contact_sub_headers = $crawler->filter('h3');
+    $this->assertCount(1, $contact_sub_headers);
+    $this->assertEquals($name, trim($contact_sub_headers->text()));
+
+    // Add and assert body field.
+    $contact->set('oe_body', 'Body text ' . $name)->save();
+    $build = $this->contactViewBuilder->view($contact, 'full');
+    $html = $this->renderRoot($build);
+    $crawler = new Crawler($html);
+    $rendered_body = $crawler->filter('.ecl-editor');
+    $this->assertCount(1, $rendered_body);
+    $this->assertEquals("Body text $name", trim($rendered_body->text()));
+
+    // Add and assert Organisation field.
+    $contact->set('oe_organisation', "Organisation $name")->save();
+    $build = $this->contactViewBuilder->view($contact, 'full');
+    $html = $this->renderRoot($build);
+    $field_list_assert = new FieldListAssert();
+    $expected_values = [];
+    $expected_values['items'][] = [
+      'label' => 'Organisation',
+      'body' => "Organisation $name",
+    ];
+    $field_list_assert->assertPattern($expected_values, $html);
+    $field_list_assert->assertVariant('horizontal', $html);
+
+    // Add and assert Website field.
+    $contact->set('oe_website', ['uri' => "http://www.example.com/website_$name"])->save();
+    $build = $this->contactViewBuilder->view($contact, 'full');
+    $expected_values['items'][] = [
+      'label' => 'Website',
+      'body' => "http://www.example.com/website_$name",
+    ];
+    $field_list_assert->assertPattern($expected_values, $this->renderRoot($build));
+
+    // Add and assert Email field.
+    $contact->set('oe_email', "$name@example.com")->save();
+    $build = $this->contactViewBuilder->view($contact, 'full');
+    $expected_values['items'][] = [
+      'label' => 'Email',
+      'body' => "$name@example.com",
+    ];
+    $field_list_assert->assertPattern($expected_values, $this->renderRoot($build));
```

Comment:
Again why we can't add all fields (rest) and then do single rendering and assertion?
For instance, in 'Call for tenders' content type it is reasonable to do that because we doing testing statuses (upcoming/past, etc) but it would be nice to understand why are we doing that here?
First of because rendering has some influence on performance.